pkg/runtime/runtime.go

```diff
@@ -1597,28 +1641,22 @@ func (a *DaprRuntime) authSecretStoreOrDefault(resource meta.Resource) string {
 	return secretStore
 }
 
-func (a *DaprRuntime) blockUntilAppIsReady(ctx context.Context) {
+func (a *DaprRuntime) blockUntilAppIsReady(ctx context.Context) error {
 	if a.runtimeConfig.appConnectionConfig.Port <= 0 {
-		return
+		return nil
 	}
 
 	log.Infof("application protocol: %s. waiting on port %v.  This will block until the app is listening on that port.", string(a.runtimeConfig.appConnectionConfig.Protocol), a.runtimeConfig.appConnectionConfig.Port)
 
 	dialAddr := a.runtimeConfig.appConnectionConfig.ChannelAddress + ":" + strconv.Itoa(a.runtimeConfig.appConnectionConfig.Port)
 
-	stopCh := ShutdownSignal()
-	defer signal.Stop(stopCh)
 	for {
 		select {
-		case <-stopCh:
-			// Cause a shutdown
-			a.ShutdownWithWait()
-			return
+		// Return
 		case <-ctx.Done():
-			// Return
-			return
-		default:
-			// nop - continue execution
+			return ctx.Err()
+		// prevents overwhelming the OS with open connections
+		case <-time.After(time.Millisecond * 100):
```

Comment:
The difference is that now the first attempt is delayed 100ms. Maybe ove this `select` at the end of the loop?

Reply:
:+1: